reg/x86.go

```diff
@@ -39,6 +42,7 @@ var (
 	ProgramCounter = Pseudo.define(S0, 0, "PC")
 	StaticBase     = Pseudo.define(S0, 0, "SB")
 	StackPointer   = Pseudo.define(S0, 0, "SP")
+	TrueOpmask     = Pseudo.define(S0, 0, "K0") // K0 can't be used as a predicate mask
```

Comment:
Question: Is this definition useful? K0 is obviously not a pseudo-register defined/recognized by the golang assembler. But there are obvious parallels in that using K0 as a predicate value doesn't really refer to the contents of the physical register. I guess I'm still unclear on what is gained/lost by setting it up this way.

Reply:
Also, now that I'm staring at it, `TrueOpmask` may not be the best name. Maybe `DefaultOpmask`...

Reply:
At this point I don't see a reason to add this as a pseudo-register. As you say, it's not really a pseudo register in the sense that the others are. For now declaring it as a `Restricted` register seems right, as you've done already.

I foresee this being quite annoying, since Go's AVX-512 syntax has instructions taking variable number of operands depending on whether masking is used. I need to think about this a bit more.